ASTree.cpp

```diff
@@ -54,7 +54,11 @@ static void CheckIfExpr(FastStack& stack, PycRef<ASTBlock> curblock)
     if (curblock->nodes().size() < 2)
         return;
     auto rit = curblock->nodes().crbegin();
-    ++rit; // the last is "else" block, the one before should be "if" (could be "for", ...)
+    // the last is "else" block, the one before should be "if" (could be "for", ...)
+    if ((*rit)->type() != ASTNode::NODE_BLOCK ||
```

Comment:
Checks exactly what the comment says now. This had false positives for ternary ifs